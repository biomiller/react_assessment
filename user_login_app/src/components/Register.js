import React, { Component } from 'react';
import axios from "axios";

import {
    Button
} from 'reactstrap';


export class Register extends Component {


    constructor() {
        super();
        this.state = {
            email: "",
            password: "",
            password2: "",
            username: "",
            status:""
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();

        let username = e.target[0].value
        let email = e.target[1].value
        let password = e.target[2].value
        let password2 = e.target[3].value

        let newUser = {
            username: username,
            email: email,
            password: password,
            password2: password2

        }

        axios
            .post("http://localhost:5000/user/createUser", newUser)
            .then(response => {
                console.log(response);
                if (response.data.Status === "Account successfully created") {
                    this.props.userLoggedIn(username);
                    this.props.getData();
                    this.setState({
                        status:"Successfully created new user: " + username
                    })

                } else {
                    this.setState({
                        username: response.data.username,
                        email: response.data.email,
                        password: response.data.password,
                        password2: response.data.password2,
                    })
                }
            })
            .catch(err => { console.log(err) })
    }

    render() {
        return (
            <form onSubmit={this.handleSubmit}>

                <h1>Register</h1>

                <input required type="text" placeholder="Username"></input>
                <p>{this.state.username}</p>

                <input required type="email" placeholder="email"></input>
                <p>{this.state.email}</p>

                <input required type="password" placeholder="Password"></input>
                <p>{this.state.password}</p>

                <input required type="password" placeholder="Repeat Password"></input>
                <p>{this.state.password2}</p>

                <Button type="submit">Submit</Button>

                <br></br>
                <p>{this.state.status}</p>

            </form>

        );
    }
}