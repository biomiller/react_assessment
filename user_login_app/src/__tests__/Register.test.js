import React from 'react';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';

import { Register } from "../components/Register.js";


it('is Register rendered', () => {
    const component = renderer
    .create(
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    )
    .toJSON();

    expect(component).toMatchSnapshot();
});



