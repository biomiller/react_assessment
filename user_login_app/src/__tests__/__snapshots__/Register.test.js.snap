// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is Register rendered 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Register
  </h1>
  <input
    placeholder="Username"
    required={true}
    type="text"
  />
  <p>
    
  </p>
  <input
    placeholder="email"
    required={true}
    type="email"
  />
  <p>
    
  </p>
  <input
    placeholder="Password"
    required={true}
    type="password"
  />
  <p>
    
  </p>
  <input
    placeholder="Repeat Password"
    required={true}
    type="password"
  />
  <p>
    
  </p>
  <button
    aria-label={null}
    className="btn btn-secondary"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
  <br />
  <p>
    
  </p>
</form>
`;
